sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/models/ShareFileInfo.java

```diff
@@ -33,6 +35,27 @@ public ShareFileInfo(String eTag, OffsetDateTime lastModified, Boolean isServerE
         this.lastModified = lastModified;
         this.isServerEncrypted = isServerEncrypted;
         this.smbProperties = smbProperties;
+        this.posixProperties = null;
+    }
+
+    //Internal constructor to support FilePosixProperties class.
+    private ShareFileInfo(String eTag, OffsetDateTime lastModified, Boolean isServerEncrypted,
+        FileSmbProperties smbProperties, FilePosixProperties posixProperties) {
+        this.eTag = eTag;
+        this.lastModified = lastModified;
+        this.isServerEncrypted = isServerEncrypted;
+        this.smbProperties = smbProperties;
+        this.posixProperties = posixProperties;
+    }
+
+    static {
+        ShareFileInfoHelper.setAccessor(new ShareFileInfoHelper.ShareFileInfoAccessor() {
+            @Override
+            public ShareFileInfo create(String eTag, OffsetDateTime lastModified, Boolean isServerEncrypted,
+                FileSmbProperties smbProperties, FilePosixProperties posixProperties) {
```

Comment:
We can use lambda expressions here for better readability.

change it to something like this:-

static {
    ShareDirectoryPropertiesHelper.setAccessor(
        (metadata, eTag, lastModified, isServerEncrypted, smbProperties, posixProperties) -> 
            new ShareDirectoryProperties(metadata, eTag, lastModified, isServerEncrypted, smbProperties, posixProperties)
    );
}